pandas/tests/indexes/datetimes/test_tools.py

```diff
@@ -270,6 +270,23 @@ def test_to_datetime_utc_is_true(self):
         expected = pd.DatetimeIndex(data=date_range)
         tm.assert_index_equal(result, expected)
 
+    def test_to_datetime_tz_kw(self):
+        # See gh-13712
+        for tz in [None, 'US/Eastern', 'Asia/Tokyo', 'UTC']:
+            data = ['20140101 000000', '20140102 000000', '20140103 000000']
+            start = pd.Timestamp(data[0], tz=tz)
+            end = pd.Timestamp(data[-1], tz=tz)
+            date_range = pd.bdate_range(start, end)
+
+            result = pd.to_datetime(data, format='%Y%m%d %H%M%S', tz=tz)
+            expected = pd.DatetimeIndex(data=date_range)
+            tm.assert_numpy_array_equal(result.values, expected.values)
```

Comment:
compare index using ``tm.assert_index_equal``

Reply:
@jreback for some context, the reason I suggested to do it like this is because assert_index_equal fails because timezone localization on single timestamp or on datetimeindex results in a slightly different tz object. This is a separate issue as the the one solved in this pr, therefore suggested this wirkaround which at least test this is correct (but didn't have time to open an issue, and from phone now)

Reply:
we never compare like this - too easy to get wrong; if there is an issue pls show

Reply:
I can't test it now, but if I remember correctly: compare pd.Timestamp('2012-01-01', tz='US/Eastern').tz with pd.to_datetime(['2012-01-01']).tz_localize('US/Eastern').tz. Both attributes are not equal